libpod/oci_conmon_linux.go

```diff
@@ -1055,6 +1055,16 @@ func (r *ConmonOCIRuntime) createOCIContainer(ctr *Container, restoreOptions *Co
 		}
 	}
 
+	if l, ok := os.LookupEnv("LISTEN_FDS"); ok {
+		listenFds, err := strconv.ParseUint(l, 10, 64)
+		if err != nil {
+			logrus.Warnf("Error LISTEN_FDS environment variable for %s not an int %v", ctr.ID(), err)
+		}
+
+		if uint(listenFds) > ctr.config.PreserveFDs {
+			ctr.config.PreserveFDs = uint(listenFds)
+		}
```

Comment:
```
		if uint(listenFds) > ctr.config.PreserveFDs {
			ctr.config.PreserveFDs = uint(listenFds)
		}
```
seems to be 
(pseudo code)
```
ctr.config.PreserveFDs := max(listenFds, ctr.config.PreserveFDs)
```

I don't understand this code. Why should it be a max value?

Reply:
I am just guessing here, that if the user specified --preserve_fds=1 and LISTEN_FDS=2, then we need to leak at least 2 FDs into the container.  But I am not really sure what should be the behaviour.  Perhaps we should ignore the LISTEN_FDS and force the user to tell us how many FDS to leak.

Reply:
At first sight runc and crun are doing the same:
See the runc example:
```
 LISTEN_PID=$pid_of_runc LISTEN_FDS=3 runc run --preserve-fds 5 <container>
```

from https://github.com/opencontainers/runc/blob/master/docs/terminals.md

-------------


Quote: _basically, it can just leave the FDs and $LISTEN_FDS untouched_
https://systemd.io/CONTAINER_INTERFACE/

I guess runc and crun could have ignored LISTEN_FDS and just make it the responsibility of 
the caller to specify a --preserve-fds  argument that includes the LISTEN_FDS.

But that is not how it is done right now.

Reply:
So we should do nothing with them except leak them into the OCI Runtimes.  Will crun and runc rename the LISTEN_PID to pid1?

Reply:
That's correct.  crun and runc preserve the FDS but only runc sets LISTEN_PID=1 (Cc: @giuseppe @flouthoc).